pkg/orchestrator/scheduler/batch_job_test.go

```diff
@@ -303,6 +304,41 @@ func (s *BatchJobSchedulerTestSuite) TestProcess_ShouldMarkJobAsFailed_NoRetry()
 	s.Require().NoError(s.scheduler.Process(ctx, evaluation))
 }
 
+func (s *BatchJobSchedulerTestSuite) TestProcess_ShouldStopExpiredExecutions() {
+	ctx := context.Background()
+	job, executions, evaluation := mockJob()
+	// Set the job type to batch and the timeout to 60 minutes
+	job.Task().Timeouts.ExecutionTimeout = int64((60 * time.Minute).Seconds())
+	// Set the start time of the executions to exceed the timeout
+	for i := range executions {
+		executions[i].ModifyTime = time.Now().Add(-90 * time.Minute).UnixNano()
```

Comment:
further motivation to adopt github.com/benbjohnson/clock in our testing.

Reply:
answered above

Reply:
using clocks in https://github.com/bacalhau-project/bacalhau/pull/3974/commits/da4a3ae81130b423b822fcbdf67ed7d97d1e10cf